relay-redis/src/real.rs

```diff
@@ -34,28 +35,68 @@ enum ConnectionInner<'a> {
     Single(&'a mut redis::Connection),
 }
 
-impl ConnectionLike for ConnectionInner<'_> {
-    fn req_packed_command(&mut self, cmd: &[u8]) -> redis::RedisResult<redis::Value> {
+impl ConnectionInner<'_> {
+    fn get_connections(&mut self) -> Vec<&mut (dyn ConnectionLike + Send)> {
         match self {
-            ConnectionInner::Cluster(ref mut con) => con.req_packed_command(cmd),
+            ConnectionInner::Cluster(con) => vec![con as &mut (dyn ConnectionLike + Send)],
             ConnectionInner::MultiWrite {
                 primary: primary_connection,
                 secondaries: secondary_connections,
             } => {
-                let primary_result = primary_connection.req_packed_command(cmd);
+                let mut connections = primary_connection.get_connections();
                 for secondary_connection in secondary_connections.iter_mut() {
-                    if let Err(error) = secondary_connection.req_packed_command(cmd) {
+                    connections.extend(secondary_connection.get_connections());
+                }
+
+                connections
+            }
+            ConnectionInner::Single(con) => vec![con as &mut (dyn ConnectionLike + Send)],
+        }
+    }
+}
+
+impl ConnectionLike for ConnectionInner<'_> {
+    fn req_packed_command(&mut self, cmd: &[u8]) -> redis::RedisResult<redis::Value> {
+        let mut results = thread::scope(|s| {
+            let handles: Vec<_> = self
+                .get_connections()
+                .into_iter()
+                .map(|connection| s.spawn(move || connection.req_packed_command(cmd)))
+                .collect();
+
+            handles
+                .into_iter()
+                .map(|handle| handle.join())
```

Comment:
Yes this is expected, from my understanding we wanted to parallelize the sending of cmds but still wait for all of them to complete. Otherwise if we do fire and forget we are hoping that secondaries are consistent with the primary.

Reply:
Nice, I read on the docs a week ago about this but forgot to use it! Will do now, thanks!

Reply:
I'm not sure if the borrowing advantage justifies waiting on a (potentially slower) secondary, but I'll leave it up to you.